src/module/encounter/combatant.ts

```diff
@@ -36,6 +37,24 @@ class CombatantPF2e<
         return this.parent.getCombatantWithHigherInit(this, than) === this;
     }
 
+    /** Get the active Combatant for the given actor, creating one if necessary */
+    static async fromActor(actor: ActorPF2e): Promise<CombatantPF2e<EncounterPF2e> | null> {
+        if (!game.combat) {
+            ui.notifications.error(game.i18n.localize("PF2E.Encounter.NoActiveEncounter"));
+            return null;
+        }
+        const token = actor.getActiveTokens().pop();
+        const existing = game.combat.combatants.find((combatant) => combatant.actor === actor);
+        if (existing) {
+            return existing;
+        } else if (token) {
+            await token.toggleCombat(game.combat);
```

Comment:
Can we use `game.combat.createEmbeddedDocuments` here so that we can control rendering? Not rendering would be desirable in some callers' cases.

Reply:
Updated. I've set it to render by default.

Reply:
Also added the `sceneId` to the `Combatant` source. It seems to work without it but core `TokenLayer#toggleCombat` uses it so it's probably safer to include it.